SemiLeptonicWVA/Utils/src/GenEventInfo.cc

```diff
@@ -264,6 +270,23 @@ GenEventInfo::produce(edm::Event& iEvent, const edm::EventSetup& iSetup)
 	  PUWeight = LumiWeights_.weight( Tnpv );
 	}
 
+
+
+	edm::Handle< LHEEventProduct > LHEEventInfo;
+        iEvent.getByLabel("externalLHEProducer", LHEEventInfo);
+
+	std::vector<float> genweights;
+        float originalWeight;
+
+	if (LHEEventInfo.isValid()){
+	  auto weightsTemp = LHEEventInfo ->weights();
+	  originalWeight *= LHEEventInfo ->originalXWGTUP();
+	  for (unsigned int i = 0; i < weightsTemp.size(); i++){
+	    genweights.push_back(weightsTemp.at(i).wgt);
+	  }
+	}
+
+
 	std::auto_ptr<double> htpw(new double(PUWeight));
```

Comment:
Just like here, where there is an 'auto_ptr' and placing the value in the iEvent, you likewise will need to place the values in the event for later recall.  Add the following lines:

        std::auto_ptr<double> owp(new double(originalWeight));
        iEvent.put(owp,"originalWeight");

        std::auto_ptr<std::vector<double>> gwp(new std::vector<double>(AQGCweights));
        iEvent.put(gwp,"AQGCweights");